packages/ann/ann/c_src/logistic_actf_component.cc

```diff
@@ -58,4 +59,76 @@ namespace ANN {
     buffer.printf("ann.components.actf.logistic{ name='%s' }", name.c_str());
     return buffer.to_string(buffer_list::NULL_TERMINATED);
   }
+
+  ///////////////////////////////////////////////////////////////////////
+
+  SparseLogisticActfANNComponent::
+  SparseLogisticActfANNComponent(const char *name,
+                                 float beta, float target_avg_act) :
+    LogisticActfANNComponent(name) {
+    this->beta = beta; this->target_avg_act = target_avg_act;
+  } 
+  
+  SparseLogisticActfANNComponent::~SparseLogisticActfANNComponent() {
+  }
+
+  ANNComponent *SparseLogisticActfANNComponent::clone() {
+    SparseLogisticActfANNComponent *obj = new
+      SparseLogisticActfANNComponent(name.c_str(), beta, target_avg_act);
+    return obj;
+  }
+
+  char *SparseLogisticActfANNComponent::toLuaString() {
+    buffer_list buffer;
+    buffer.printf("ann.components.actf.sparse_logistic{ "
+                  "name='%s', beta=%f, rho=%f  }",
+                  name.c_str(), this->beta, this->target_avg_act);
+    return buffer.to_string(buffer_list::NULL_TERMINATED);
+  }
+ 
+  void SparseLogisticActfANNComponent::multiplyDerivatives(MatrixFloat *input_units,
+                                                           MatrixFloat *output_units,
+                                                           MatrixFloat *input_errors,
+                                                           MatrixFloat *output_errors) {
+    UNUSED_VARIABLE(input_units);
+    
+    const float INV_DIM0 = 1.0f/static_cast<float>(output_units->getDimSize(0));
+    const float EPS = AprilMath::Limits<float>::epsilon();
+    // Computing average activations
+    MatrixFloat *current_avg = matSum(output_units, 0);
+    matScal(current_avg, INV_DIM0);
+     
+    // -p/\hat{p}
+    MatrixFloat *sparse_gradients = current_avg->clone();
+    matClamp(sparse_gradients, EPS, 1.0f);
+    matDiv(sparse_gradients, -target_avg_act);
+    
+    // (1-p)/(1-\hat{p})
+    MatrixFloat *aux = current_avg; // reusing current_avg
+    matComplement(aux);
+    matClamp(aux, EPS, 1.0f);
+    matDiv(aux, (1.0f-target_avg_act));
+    
+    // -p/\hat{p} + (1-p)/(1-\hat{p} = sparse_gradients + aux
+    matAxpy(sparse_gradients, 1.0f, aux);
+    
+    // Normalize the error
+    // matScal(sparse_gradients, INV_DIM0);
```

Comment:
Do we need to normalize by the bunch size? I have a big doubt around this issue :S